File: back/src/main/java/com/back/domain/waiting/dto/WaitingQueueResponse.java
package com.back.domain.waiting.dto;

public record WaitingQueueResponse(
        Long concertId,
        Long scheduleId,
        Long userId,
        Long rank,
        Long myQueueNumber,
        String entryToken
) {
    public static WaitingQueueResponse of(Long concertId, Long scheduleId, Long userId, Long rank, Long myQueueNumber, String entryToken) {
        return new WaitingQueueResponse(concertId, scheduleId, userId, rank, myQueueNumber, entryToken);
    }
}

File: back/src/main/kotlin/com/back/domain/waiting/dto/WaitingQueueResponse.kt
package com.back.domain.waiting.dto

data class WaitingQueueResponse(
    val concertId: Long,
    val scheduleId: Long,
    val userId: Long,
    val rank: Long,
    val myQueueNumber: Long,
    val entryToken: String?
) {
    companion object {
        fun of(
            concertId: Long,
            scheduleId: Long,
            userId: Long,
            rank: Long,
            myQueueNumber: Long,
            entryToken: String?
        ): WaitingQueueResponse {
            return WaitingQueueResponse(concertId, scheduleId, userId, rank, myQueueNumber, entryToken)
        }
    }
}
